xbmc/cores/RetroPlayer/RetroPlayer.cpp

```diff
@@ -357,14 +358,9 @@ void CRetroPlayer::SetSpeed(float speed)
 
     if (m_gameClient->GetPlayback()->GetSpeed() != 0.0)
       CloseOSD();
-  }
-}
 
-float CRetroPlayer::GetSpeed()
-{
-  if (m_gameClient)
-    return static_cast<float>(m_gameClient->GetPlayback()->GetSpeed());
-  return 0;
+    CDataCacheCore::GetInstance().SetSpeed(1.0, speed);
```

Comment:
@garbear btw: I won't use the generic IPlayer interface for this case where you know the type. you can implement a GetSpeed method for RP if you need one. GetSpeed in IPlayer is gone